internal/gitprovider/gitlab/gitlab.go

```diff
@@ -27,8 +27,19 @@ var (
 			// e.g. 'git.mycompany.com'
 			return strings.Contains(u.Host, GitProviderServiceName)
 		},
-		NewService: func() (gitprovider.GitProviderService, error) {
-			return NewGitLabProvider()
+		NewService: func(repoURL string) (gitprovider.GitProviderService, error) {
+			if repoURL != "" {
+				u, err := url.Parse(repoURL)
+				if err != nil {
+					return nil, err
+				}
+				u.Path = ""
+				u.RawQuery = ""
+				u.Fragment = ""
+				return NewGitLabProvider(u.String() + "/api/v4")
+			} else {
+				return NewGitLabProvider("")
+			}
```

Comment:
I feel things would be a bit tidier if you just passed `repoURL` to `NewGitLabProvider()` and let it do the parsing.